custom_components/adaptive_lighting/switch.py

```diff
@@ -553,7 +553,8 @@ def _expand_light_groups(hass: HomeAssistant, lights: list[str]) -> list[str]:
 
 def _supported_features(hass: HomeAssistant, light: str):
     state = hass.states.get(light)
-    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    # see #423, I guess not all lights have ATTR_SUPPORTED_FEATURES
+    supported_features = state.attributes.get(ATTR_SUPPORTED_FEATURES, {})
```

Comment:
I think this is incorrect. `state.attributes[ATTR_SUPPORTED_FEATURES]` is an `enum.IntFlag` which is a subtype of `int`. So I think the correct thing is to do `state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)`

Reply:
Please take a look at the issue commented above that change.

Reply:
I see what that person reported, but in the case that `ATTR_SUPPORTED_FEATURES` is indeed missing. The following happens:
```python
supported_features = {}
value = 1
supported_features & value
```
which will raise:
```
>>> supported_features & value
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for &: 'dict' and 'int'
```

Reply:
Hmm... interesting. [I just compared to the older releases](https://github.com/basnijholt/adaptive-lighting/compare/1.3.0...1.6.0) that released around the time that he posted, and confirmed no code changed there...

Your check is great! I'm just wondering what happened to that person.